### ISSUE TITLE
Add the option to not use/load KaTeX.js

### ISSUE THREAD

**Issue by vitaly-zdanevich** [halogenica/beautifulhugo#163]
Because it affect loading speed and 99% of blogs do not render math.

Or maybe in Hugo it is possible to inject this library only when needed - according to posts scanning result.

**Comment by halogenica** [comment]
Great idea, I'll try to add a smart way to configure this. If anyone has a PR I'd love to merge it.

**Comment by clux** [comment]
Oh i misread this issue. My pr does a much dumber thing :(
Not sure how to detect it easily - it would be particularly hairy if people want to customise their delimiters.

It'd probably be better to have a flag in each post header when needed (i.e. along with tags and categories), set `latex = true` there, then if true, call an overrideable partial. Maybe.

**Comment by VincentTam** [comment]
> It'd probably be better to have a flag in each post header when needed (i.e. along with tags and categories), set `latex = true` there, then if true, call an overrideable partial. Maybe.

The backward compatibility of existing blogs with KaTeX loaded is worth considering.  We _can't_ just break the math rendering by a `git submodule update` _without_ giving, at least, a warning :warning: --- but that would be impossible when using Git in a shell.

> Or maybe in Hugo it is possible to inject this library only when needed - according to posts scanning result.

@vitaly-zdanevich [Mmark](https://gohugo.io/content-management/formats/#mmark) does the [job](https://github.com/mmarkdown/mmark/search?q=math&unscoped_q=math) of converting `$$ \sin(x) $$` into `\( \sin(x) \)`, and

```markdown
$$
\sin(x) = 1
$$
```

as a block `<div>` element, but you might reinvent the wheel if you really want to.

**Comment by DoktorMike** [comment]
Also slightly related but is it currently not possible to do inline math? I can render the block math just fine but I cannot seem to do the inline version.

**Comment by StevenDoesStuffs** [comment]
Yeah, KaTeX doesn't like inline math. I think we should use mathjax instead. @halogenica would you be willing to merge a PR which replaces katex with mathjax instead, and adds a config option to enable it?

**Comment by halogenica** [comment]
I’m always happy to look over PRs and I think adding another integration with a config option to enable it would be great! I can’t say exactly when I’ll get to integrating it, but as long as it’s a config option it should be easy :)

**Comment by StevenDoesStuffs** [comment]
@halogenica I've just submitted #345. It's very small, so hopefully it wont be that hard to review.
